lib/chat_web/live/main_live/page/dialog.ex

```diff
@@ -114,7 +114,9 @@ defmodule ChatWeb.MainLive.Page.Dialog do
         end
       )
 
-    FileIndex.add_file(chunk_key, dialog)
+    {_index, msg} = message
+
+    FileIndex.add_file(chunk_key, dialog, msg.id, chunk_secret)
```

Comment:
We SHOULD NOT store files secrets unprotected. We SHOULD encrypt it with user public key.

1. Why do you need `secret` here. I do not see it used somewhere?
2. Is `Upload.secret` protected, so that no one but user can read partially uploaded  file?

Reply:
It starts with the check of whether the file already exists:
https://github.com/Buckitup-chat/chat/blob/342e935ff0edd2dec0330928bbb9c993ee7120e9/lib/chat_web/live/main_live/index.ex#L427
https://github.com/Buckitup-chat/chat/blob/342e935ff0edd2dec0330928bbb9c993ee7120e9/lib/chat_web/live/main_live/index.ex#L441-L452
Then it gets used inside Dialog and Room LiveViews to create a `%File{}`
https://github.com/Buckitup-chat/chat/blob/342e935ff0edd2dec0330928bbb9c993ee7120e9/lib/chat_web/live/main_live/page/dialog.ex#L106-L111
which contains the chunk key and the secret
https://github.com/Buckitup-chat/chat/blob/13c30b28d3c7f177dad56c7580a8b0910f6b2e88/lib/chat/messages/file.ex#L8-L29
that are later used to read the chunks (not yet, these aren't the same as the chunk key and the secret)
https://github.com/Buckitup-chat/chat/blob/342e935ff0edd2dec0330928bbb9c993ee7120e9/lib/chat_web/live/main_live/layout/message.ex#L496-L506
here
https://github.com/Buckitup-chat/chat/blob/342e935ff0edd2dec0330928bbb9c993ee7120e9/lib/chat_web/live/main_live/layout/message.ex#L510-L519
https://github.com/Buckitup-chat/chat/blob/342e935ff0edd2dec0330928bbb9c993ee7120e9/lib/chat_web/controllers/file_controller.ex#L39-L45

I'd appreciate help in figuring out how to read those chunks without having the secret in the file index. Currently, the secret is there inside any reference of the file.

Reply:
Oh, I see the case when it is needed. `:file_index` is correct place. 
The problem is that secret is not encrypted. I.e. anyone could read anyone's files.

on storing it should be protected with
```elixir
secret
|> Base.encode64!()
|> Chat.Utils.encrypt(me)
```

and creating new message - uncovered. Since message itself is protected when stored in db
```elixir
secret = encrypted_secret
|> Chat.Utils.decrypt(me)
|> Base.decode64()
```

The same protection should be done to `secret` in `Upload` credentials

Reply:
Thanks!